### PR TITLE
fix(AuditTrail): allow to see action option only if user have view au…

### LINKED ISSUE #15431: AuditTrail - see action button only if user have right permission
### Describe the bug
Now `Audit trail` button for each content item, in the content item list view, is visibile by default. No related permission evaluation. 

If a user click `Audit Trail` button, without `View audit trail` permission, you see the forbid page.

Screenshot 2024-03-01 085639

### To Reproduce
1. enable audit trail module
2. go in content item list view with a user without permission to see audit trail
3. click over action button 
4. check if you see `Audit Trail` option

### Expected behavior
Allow to see `Audit Trail` button only if user have the permission to view audit trail.

see image: no audit trail menu, no audit trail option

Screenshot 2024-03-01 085858

### PR DESCRIPTION
**Pull request by PiemP**
Fixes #15431

fix(AuditTrail): allow to see action option only if user have view audit trail permission

### CODE REVIEW

**Comment by Piedone** [comment]
I won't be able to work on reviewing this.

**Comment by hishamco** [comment]
@Piedone I remember on of the Lombiq team who implement AuditTrail, is it Gabor?

File: src/OrchardCore.Modules/OrchardCore.Contents/AuditTrail/Drivers/AuditTrailContentsDriver.cs [hunk 1]
```diff
@@ -8,10 +10,22 @@ namespace OrchardCore.Contents.AuditTrail.Drivers
...
+            _httpContextAccessor = httpContextAccessor;
+            _authorizationService = authorizationService;
+        }
+
         public override IDisplayResult Display(ContentItem contentItem, IUpdateModel updater)
         {
-            return Initialize<ContentItemViewModel>("AuditTrailContentsAction_SummaryAdmin", m => m.ContentItem = contentItem).Location("SummaryAdmin", "ActionsMenu:10");
+            return Initialize<ContentItemViewModel>("AuditTrailContentsAction_SummaryAdmin", m => m.ContentItem = contentItem)
+                .Location("SummaryAdmin", "ActionsMenu:10")
+                .RenderWhen(() => _authorizationService.AuthorizeAsync(_httpContextAccessor.HttpContext?.User, OrchardCore.AuditTrail.AuditTrailPermissions.ViewAuditTrail));
```
Comment (Piedone) [inline src/OrchardCore.Modules/OrchardCore.Contents/AuditTrail/Drivers/AuditTrailContentsDriver.cs:28]: Import the namespace with `using` instead so we can have a shorter line.

**Review by Piedone — CHANGES_REQUESTED** [review]
Yep, but I found some time :).

Please request review when done.

File: src/OrchardCore.Modules/OrchardCore.Contents/AuditTrail/Drivers/AuditTrailContentsDriver.cs [hunk 2]
```diff
@@ -1,17 +1,32 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
 using OrchardCore.ContentManagement;
 using OrchardCore.ContentManagement.Display.ContentDisplay;
 using OrchardCore.ContentManagement.Display.ViewModels;
 using OrchardCore.DisplayManagement.ModelBinding;
 using OrchardCore.DisplayManagement.Views;
+using OrchardCore.AuditTrail;
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Contents/AuditTrail/Drivers/AuditTrailContentsDriver.cs:8]: Sort the using

**Review by Piedone — APPROVED** [review]

**Comment by Piedone** [comment]
@all-contributors please add @PiemP for code.

**Comment by Piedone** [comment]
Ah, just the name is "tempcode" there.

**Comment by Piedone** [comment]
Thank you for your repeated contribution!

**Comment by hishamco** [comment]
Ya, @PiemP is already a contributor AFAIK, but it's good that the bot is confirm that  ;)

**Pull request merged**